### PR TITLE
Users/mg/xavs15

### PR DESCRIPTION
**Pull request by madhurig**

### CODE REVIEW

File: Tasks/XamarinAndroid/Get-JavaDevelopmentKitPath.ps1 [hunk 1]
```diff
@@ -0,0 +1,96 @@
+function Get-JavaDevelopmentKitPath{
```
Comment (madhurig) [inline Tasks/XamarinAndroid/Get-JavaDevelopmentKitPath.ps1]: The logic is similar to how the capabilities are detected in the vsts-agent: https://github.com/Microsoft/vsts-agent/blob/master/src/Misc/layoutbin/powershell/Add-JavaCapabilities.ps1

File: Tasks/XamarinAndroid/Select-MSBuildLocation.ps1 [hunk 2]
```diff
@@ -0,0 +1,63 @@
+function Select-MSBuildLocation {
```
Comment (madhurig) [inline Tasks/XamarinAndroid/Select-MSBuildLocation.ps1]: Copy of Select-MSBuildLocation.ps1 from MSBuild task

File: Tasks/XamarinAndroid/task.json [hunk 3]
```diff
@@ -73,13 +73,56 @@
...
+            }
+        },
+        {
+            "name": "msbuildVersion",
+            "type": "pickList",
+            "label": "MSBuild Version",
+            "required": false,
+            "groupName": "msbuildOptions",
+            "defaultValue": "latest",
+            "helpMarkDown": "If the preferred version cannot be found, the latest version found will be used instead. On a Mac xbuild(Mono) will be used.",
```
Comment (DavidStaheli) [inline Tasks/XamarinAndroid/task.json]: P999: "On a Mac**, xbuild (Mono)** will be used **instead of MSBuild.**"

**Comment by DavidStaheli** [comment]
Looks good to me.

File: Tasks/XamarinAndroid/task.loc.json [hunk 4]
```diff
@@ -73,13 +73,56 @@
...
+      },
+      "visibleRule": "msbuildLocationMethod = version"
+    },
+    {
       "name": "msbuildLocation",
       "type": "string",
       "label": "ms-resource:loc.input.label.msbuildLocation",
       "defaultValue": "",
-      "required": false,
+      "required": true,
```
Comment (ericsciple) [inline Tasks/XamarinAndroid/task.loc.json]: required?

Comment (madhurig) [inline Tasks/XamarinAndroid/task.loc.json]: Yes, required if "location" radio button is picked.

File: Tasks/XamarinAndroid/XamarinAndroid.ps1 [hunk 5]
```diff
@@ -1,113 +1,83 @@
...
 
-$args = $msbuildArguments;
+    $msbuildLocation = Select-MSBuildLocation -Method $msbuildLocationMethod -Location $msbuildLocation -Version $msbuildVersion -Architecture $msbuildArchitecture
 
-if ($configuration)
-{
-    Write-Verbose "adding configuration: $configuration"
-    $args = "$args /p:configuration=$configuration"
+    # build each project file
+    Invoke-BuildTools -SolutionFiles $projectFiles -MSBuildLocation $msbuildLocation -MSBuildArguments $msBuildArguments -Clean:$clean
```
Comment (ericsciple) [inline Tasks/XamarinAndroid/XamarinAndroid.ps1]: i think the default before was `-CreateLogFile`

File: Tasks/XamarinAndroid/Get-JavaDevelopmentKitPath.ps1 [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+        [Parameter(Mandatory = $true)]
+        [ValidateSet('Default', 'Registry32', 'Registry64')]
+        [string]$View,
+
+        [Parameter(Mandatory = $true)]
+        [string]$KeyName,
+
+        [string]$ValueName)
+
+    Write-Host "Checking: hive '$Hive', view '$View', key name '$KeyName', value name '$ValueName'"
```
Comment (madhurig) [inline Tasks/XamarinAndroid/Get-JavaDevelopmentKitPath.ps1]: Done!

**Pull request merged**